File: test/EPI_CSharp_Test/Problem_5_1_Test.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Xunit;
using static EPI_CSharp.Problem_5_1;

namespace EPI_CSharp_Test
{
    public class Problem_5_1_Test
    {
        [Fact]
        public void Parity_Positive_Success()
        {
            // arrange
            var value = 0x753L;
            var expect = true;

            // act
            var result = Parity(value);

            // assert
            Assert.Equal(expect, result);
        }

        [Fact]
        public void Parity_NegativeValue_Success()
        {
            // arrange
            var value = -0x753L;
            var expect = false;


            // act
            var result = Parity(value);

            // assert
            Assert.Equal(expect, result);
        }

        [Fact]
        public void Parity_MinLongValue_Success()
        {
            // arrange
            var value = Int64.MinValue;
            var expect = true;

            //act
            var result = Parity(value);

            // assert
            Assert.Equal(expect, result);
        }
    }
}

File: test/EPI_Java_Test/epi/Problem_5_1_Test.java
package epi;

import org.junit.Assert;
import org.junit.Test;

import static epi.Problem_5_1.parity;

/**
 * Created by rick on 2017/1/16.
 */
public class Problem_5_1_Test {
    @Test
    public void parity_PositiveValue_Success() {
        // arrange
        long value = 0x753L;
        boolean expect = true;

        // act
        boolean result = parity(value);

        // assert
        Assert.assertEquals(expect, result);
    }

    @Test
    public void parity_NegativeValue_Success() {
        // arrange
        long value = -0x753L;
        boolean expect = false;

        // act
        boolean result = parity(value);

        //assert
        Assert.assertEquals(expect, result);
    }

    @Test
    public void parity_MinNegativeValue_Success() {
        // arrange
        long value = Long.MIN_VALUE;

        boolean expect = true;

        // act
        boolean result = parity(value);

        // assert
        Assert.assertEquals(expect, result);
    }
}
